### ISSUE TITLE
Document viewer | Subject line interaction

### ISSUE THREAD

**Issue by gnakm** [department-of-veterans-affairs/caseflow#3687]

**Comment by dannysellers** [comment]
Where does this data come from? Or is the textbox editable?

**Comment by dannysellers** [comment]
Where does this data come from? Or is the textbox editable?

**Comment by gnakm** [comment]
@dannysellers textbox is freeform!

**Comment by dannysellers** [comment]
👍

**Comment by mdbenjam** [comment]
@abbyraskinUSDS do we think we will want to know who created the document description. At some point would we want to show different descriptions for different users?

**Comment by mdbenjam** [comment]
@gnakm why is document description not like the other editable fields with an edit button? When do we want this to save, just on blur? What happens if a user types, doesn't blur the field, and the refreshes the page. Should it save?

**Comment by abbyraskinUSDS** [comment]
> At some point would we want to show different descriptions for different users?

@mdbenjam @gnakm good q. on the front end, i think we'd want to handle this scenario similarly to the way we intend to handle user-specific comments/toggling in the future. that said, i'm not sure about priority here yet - let's talk at story time. @gnakm we should think about how we might fit doc descriptions into the user toggle interaction you designed (if we determine this is a priority).

> do we think we will want to know who created the document description. 

On the metrics side, i'd be curious to look at how many unique users are using the feature. per above, I think we should probably store this information for now and figure out what is actually necessary to show to the user (and when)?

**Comment by dannysellers** [comment]
@gnakm Yeah, we certainly could add an Edit -> Cancel/Save flow for this—I assume the design of the editing state would be like a compact version of the edit annotation design?

<img width="325" alt="screen shot 2018-01-02 at 1 05 28 pm" src="https://user-images.githubusercontent.com/8533004/34494263-a831c9a2-efbd-11e7-812d-2abdd3ec7bc2.png">

Also, is there any question of who should see the links (permissions)?

**Comment by gnakm** [comment]
@dannysellers So the design that I have above does not yet exist yet but we will need it for future stuff like editing document type and receipt date #1997 and #2745. We went with this styling because it saves space at the top.

**Comment by dannysellers** [comment]
Got it. I like the text buttons, they definitely do save space. So I'll go ahead and give this field an Edit -> Save flow with the styling from those tickets?

**Comment by dannysellers** [comment]
@gnakm 

Viewing:
<img width="337" alt="screen shot 2018-01-03 at 4 33 40 pm" src="https://user-images.githubusercontent.com/8533004/34540825-fcdfbcc0-f0a3-11e7-89d4-62eba4d35800.png">

Editing:
<img width="332" alt="screen shot 2018-01-03 at 4 33 47 pm" src="https://user-images.githubusercontent.com/8533004/34540830-008d16ce-f0a4-11e7-8fd2-1e6566d86fd8.png">

Error:
<img width="359" alt="screen shot 2018-01-03 at 4 42 30 pm" src="https://user-images.githubusercontent.com/8533004/34541140-20666094-f0a5-11e7-867c-a09fe10e1712.png">

This is now in a new `EditableDocumentField` component, so we can easily give other fields the same flow @mdbenjam

**Comment by gnakm** [comment]
@dannysellers This looks great and yaay we can reuse these for later! I'll make tickets to add "save text inputs" for style guide. Two things. I added this in the other issues but forgot to add it here. Could you take out the semi-colon? Now that edit is next to the title, the colons change the context. Also, can you add 20px spacing btwn each section? Right now, every line is equally spaced and it's hard to scan them by sections. 

![image](https://user-images.githubusercontent.com/13420618/34569114-bd452b58-f135-11e7-8d70-fd69c9637ce8.png)

**Comment by dannysellers** [comment]
👍 

![screen shot 2018-01-04 at 1 10 21 pm](https://user-images.githubusercontent.com/8533004/34577839-a8ea83f4-f150-11e7-90dd-a99b19b47b14.png)
